### ISSUE TITLE
Add support for `yarn upgrade [package...]`

### ISSUE THREAD

**Issue by cpojer** [yarnpkg/yarn#598]
Seems like this is missing or got lost somewhere. Nbd for now.

**Comment by AlicanC** [comment]
It would be nice if we had an [npm-check](https://github.com/dylang/npm-check) or [npm-upgade](https://github.com/th0r/npm-upgrade) like CLI when this command is ran without a package name.

**Comment by fab1an** [comment]
I would like to know the workflow for this as well, I'm using `npm-check` and the moment and I'm not sure how to use it along with `yarn`.

**Comment by shnhrrsn** [comment]
FWIW the [docs](https://yarnpkg.com/en/docs/usage) currently imply this works:

<img width="371" alt="screen shot 2016-10-11 at 1 55 19 pm" src="https://cloud.githubusercontent.com/assets/42901/19281969/68861d12-8fba-11e6-9026-38afbbc7debc.png">

So running was confusing:

```bash
$ yarn upgrade package
yarn upgrade v0.15.1
error This command doesn't require any arguments.
info Visit http://yarnpkg.com/en/docs/cli/upgrade for documentation about this command.
```

**Comment by kittens** [comment]
Fixed via #1077. I knew I added support for this so was confused why it wasn't working. Thanks @wyze!

**Comment by FezVrasta** [comment]
Is it normal that when I run `yarn upgrade <package>`, unrelated dependencies in`yarn.lock` are upgraded as well?
